edb/ir/typeutils.py

```diff
@@ -398,6 +399,13 @@ def ptrref_from_ptrcls(
 
     ircls: Type[irast.BasePointerRef]
 
+    source_ref: Union[Optional[irast.TypeRef],
```

Comment:
I'm not sure this is correct. The `_ref` variables cannot be anything other than a `TypeRef`.  I suggest flipping the `if isinstance(target, s_type.Type)` check below to check for `TypeRef` and adding `assert isinstance(s_types.Type)`

```python
   if not isinstance(target, irast.TypeRef):
        assert isinstance(target, s_types.Type)
        target_ref = type_to_typeref(schema, target, cache=typeref_cache)
    else:
        target_ref = target
```

Reply:
If I apply this change, an internal server error happens while populating the test database:

```
edb test -j4
Collected 2974/2974 tests.
Using up to 4 processes to run tests.
Populating test databases... Task exception was never retrieved
future: <Task finished name='Task-142' coro=<<coroutine without __name__>()> exception=ConnectionAbortedError()>
Traceback (most recent call last):
  File "edgedb/protocol/protocol.pyx", line 798, in connect
  File "edgedb/protocol/asyncio_proto.pyx", line 60, in wait_for_message
ConnectionAbortedError


Shutting down test cluster... OK.
Traceback (most recent call last):
  File ".../venv/bin/edb", line 11, in <module>
    load_entry_point('edgedb-server', 'console_scripts', 'edb')()
  File ".../venv/lib/python3.8/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File ".../venv/lib/python3.8/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File ".../venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File ".../venv/lib/python3.8/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File ".../venv/lib/python3.8/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File ".../edgedb/edb/tools/test/__init__.py", line 153, in test
    result = run()
  File ".../edgedb/edb/tools/test/__init__.py", line 265, in _run
    result = test_runner.run(suite)
  File ".../edgedb/edb/tools/test/runner.py", line 690, in run
    tb.setup_test_cases(
  File ".../edgedb/edb/testbase/server.py", line 838, in setup_test_cases
    return asyncio.run(_run())
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 612, in run_until_complete
    return future.result()
  File ".../edgedb/edb/testbase/server.py", line 835, in _run
    g.create_task(controller(
  File ".../edgedb/edb/common/taskgroup.py", line 169, in __aexit__
    raise me from None
edb.common.taskgroup.TaskGroupError: unhandled errors in a TaskGroup; 3 sub errors: (InternalServerError)
 + InternalServerError: 
 |   File ".../edgedb/edb/testbase/server.py", line 832, in controller
 |     await coro(*args)
 |   File ".../edgedb/edb/testbase/server.py", line 861, in _setup_database
 |     await dbconn.execute(setup_script)
 |   File ".../edgedb-python/edgedb/asyncio_con.py", line 165, in execute
 |     await self._protocol.simple_query(query)
 |   File "edgedb/protocol/protocol.pyx", line 497, in simple_query


 + InternalServerError: 
 |   File ".../edgedb/edb/testbase/server.py", line 832, in controller
 |     await coro(*args)
 |   File ".../edgedb/edb/testbase/server.py", line 861, in _setup_database
 |     await dbconn.execute(setup_script)
 |   File ".../edgedb-python/edgedb/asyncio_con.py", line 165, in execute
 |     await self._protocol.simple_query(query)
 |   File "edgedb/protocol/protocol.pyx", line 497, in simple_query


 + InternalServerError: 
 |   File ".../edgedb/edb/testbase/server.py", line 832, in controller
 |     await coro(*args)
 |   File ".../edgedb/edb/testbase/server.py", line 861, in _setup_database
 |     await dbconn.execute(setup_script)
 |   File ".../edgedb-python/edgedb/asyncio_con.py", line 165, in execute
 |     await self._protocol.simple_query(query)
 |   File "edgedb/protocol/protocol.pyx", line 497, in simple_query

```

By the way, is there a way to get more information from the test server, like a full traceback?